dags/atd_knack_traffic_detectors_weekly_snapshot.py

```diff
@@ -0,0 +1,120 @@
+# test locally with: docker compose run --rm airflow-cli dags test atd_knack_traffic_detectors_weekly_snapshot
+
+import os
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+from pendulum import datetime, duration
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "execution_timeout": duration(minutes=10),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+REQUIRED_SECRETS = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"production.apiKey",
+    },
+    "SOCRATA_API_KEY_ID": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeyId",
+    },
+    "SOCRATA_API_KEY_SECRET": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeySecret",
+    },
+    "SOCRATA_APP_TOKEN": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.appToken",
+    },
+    "PGREST_ENDPOINT": {
+        "opitem": "atd-knack-services PostgREST",
+        "opfield": "production.endpoint",
+    },
+    "PGREST_JWT": {
+        "opitem": "atd-knack-services PostgREST",
+        "opfield": "production.jwt",
+    },
+    "AWS_ACCESS_ID": {
+        "opitem": "Socrata Dataset Backups S3 Bucket",
+        "opfield": "production.AWS Access Key",
+    },
+    "AWS_SECRET_ACCESS_KEY": {
+        "opitem": "Socrata Dataset Backups S3 Bucket",
+        "opfield": "production.AWS Secret Access Key",
+    },
+    "BUCKET": {
+        "opitem": "Socrata Dataset Backups S3 Bucket",
+        "opfield": "production.Bucket",
+    },
+}
+
+
+with DAG(
+    dag_id="atd_knack_traffic_detectors_weekly_snapshot",
+    description="Appends traffic detector assets to a running log in Socrata",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="0 1 * * SUN" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```

Comment:
Runs 1AM on Sundays
[0 1 * * SUN](https://crontab.guru/#0_1_*_*_SUN)

Reply:
@ChristinaTremel is this a good time?

Reply:
yes, i think avoiding the weekends is a good call. do you want to add retries? i see you have it set to `0` at the moment.

Reply:
> Uppercase, and lowercase or mixed-case three character strings, based on the English name of the day: mon, tue, wed, thu, fri, sat, or sun.

Never in 25 years have I seen this, and it's widely supported. Works for months too, `Jan`, `Oct`, `Dec`, etc. I have struggled keeping the numbers of months straight for my entire life, and this end-runs this. TIL of the year.

Reply:
sounds good, moved it to 1AM Monday. And added retries to hopefully have less gaps in data!